airflow/contrib/hooks/livy_hook.py

```diff
@@ -0,0 +1,379 @@
+        self.log.info("Batch submitted with session id: {}".format(batch_id))
+
+        return batch_id
+
+    def get_batch(self, session_id):
+        """
+        Fetch info about the specified batch
+
+        :param session_id: identifier of the batch sessions
+        :type session_id: int
+        :return: response body
+        :rtype: dict
+        """
+        self._validate_session_id(session_id)
+
+        self.log.debug("Fetching info for batch session {}".format(session_id))
+        response = self.run_method(endpoint='/batches/{}'.format(session_id))
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Unable to fetch batch with id: {}".format(session_id))
+
+        return response.json()
+
+    def get_batch_state(self, session_id):
+        """
+        Fetch the state of the specified batch
+
+        :param session_id: identifier of the batch sessions
+        :type session_id: Union[int, str]
+        :return: batch state
+        :rtype: str
+        """
+        self._validate_session_id(session_id)
+
+        self.log.debug("Fetching info for batch session {}".format(session_id))
+        response = self.run_method(endpoint='/batches/{}/state'.format(session_id))
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Unable to fetch state for batch id: {}".format(session_id))
+
+        jresp = response.json()
+        if 'state' not in jresp:
+            raise AirflowException("Unable to get state for batch with id: {}".format(session_id))
+        return BatchState(jresp['state'])
+
+    def delete_batch(self, session_id):
+        """
+        Delete the specified batch
+
+        :param session_id: identifier of the batch sessions
+        :type session_id: int
+        :return: response body
+        :rtype: dict
+        """
+        self._validate_session_id(session_id)
+
+        self.log.info("Deleting batch session {}".format(session_id))
+        response = self.run_method(
+            method='DELETE',
+            endpoint='/batches/{}'.format(session_id)
+        )
+
+        if response.status_code != 200:
+            self.log.warning("Got status code {} for session {}".format(response.status_code, session_id))
+            raise AirflowException("Could not kill the batch with session id: {}".format(session_id))
+
+        return response.json()
+
+    @staticmethod
+    def _validate_session_id(session_id):
+        """
+        Validate session id is a int
+
+        :param session_id: session id
+        :type session_id: Union[int, str]
+        """
+        try:
+            int(session_id)
+        except (TypeError, ValueError):
+            raise AirflowException("'session_id' must represent an integer")
```

Comment:
```suggestion
            raise TypeError("'session_id' must represent an integer")
```

(AirflowException doesn't have any special meaning, and a TypeError more accurately describes the problem)